vnext/Microsoft.ReactNative.Cxx/JSValue.cpp

```diff
@@ -3,9 +3,324 @@
 
 #include "pch.h"
 #include "JSValue.h"
+#include <iomanip>
+#include <sstream>
+#include <string_view>
 
 namespace winrt::Microsoft::ReactNative {
 
+//===========================================================================
+// JSValue type conversion helpers.
+//===========================================================================
+
+namespace {
+
+struct NullConverter {
+  static constexpr char const *NullString = "null";
+};
+
+struct ObjectConverter {
+  static constexpr char const *JavaScriptString = "[object Object]";
+};
+
+struct StringConverter {
+  static std::string_view TrimString(std::string const &value) noexcept {
+    constexpr char const *WhiteSpace = " \n\r\t\f\v";
+
+    size_t start = value.find_first_not_of(WhiteSpace);
+    if (start == std::string::npos) {
+      return "";
+    }
+
+    size_t end = value.find_last_not_of(WhiteSpace);
+    return {value.data() + start, end - start + 1};
+  }
+
+  static std::ostream &WriteAsJsonString(std::ostream &stream, std::string const &value) noexcept {
```

Comment:
>std::ostream [](start = 9, length = 12)

What's the point of returning stream, if all this function does is modify existing one

Reply:
This is just a common technique that allows call chaining and also helps me using 'return' in 'switch' statement to make code shorter in some other places.

---
In reply to: [389199709](https://github.com/microsoft/react-native-windows/pull/4260#discussion_r389199709) [](ancestors = 389199709)